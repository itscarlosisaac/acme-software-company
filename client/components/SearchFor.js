import React, { Component, Fragment } from 'react'
import GifServices from '../services/gifServices';
import EventEmitter from '../events/emitter';

export default class SearchFor extends Component {
  state = {
    value: '',
    gifs: [],
    gifCount: 0
  }

  handleSubmit = (e) => {
    e.preventDefault();
    GifServices.FecthData(this.state.value, 5, 0, (data) => {
      this.setState( (prevState) => {
        EventEmitter.emit('LoadGifs', data)
        return { 
            gifs: data,
            gifCount: data.length
          }
      })
    })
  }

  handleChange = (event) => {
    const val = event.target.value;
    this.setState({value: val})
  }

  render() {
    return (
      <Fragment>
        <form onSubmit={this.handleSubmit}>
          <input type="text" value={this.state.value} onChange={this.handleChange} placeholder="Search for a gif" />
        </form>
      </Fragment>
    )
  }
}
