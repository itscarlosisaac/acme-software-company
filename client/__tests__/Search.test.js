import React from 'react';
import SearchFor from '../components/SearchFor';

import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter() });

describe(`<SearchFor/> - Search Component`, () => {

  it(`Renders the component`, () => {
    const wrapper = shallow(<SearchFor />);
    expect(wrapper.exists()).toBe(true);
  });

  it('On input should changes the state of the component', () => {
    const wrapper = mount(<SearchFor/>);
    const input = wrapper.find('input');
    expect(wrapper.state().value).toBe("");
    input.simulate('change', {target: { value: "Jackie"}});
    expect(wrapper.state().value).toBe("Jackie")
  });

  it(`Prevents the form from being submitted`, () => {
    const wrapper = shallow(<SearchFor />);
    let prevented = false;
    wrapper.find('form').simulate("submit", {
      preventDefault: () => prevented = true
    });
    expect(prevented).toBe(true)
  })

  it('On Submit should update states gif count and gifs array', () => {
    const wrapper = mount(<SearchFor/>);
    const form = wrapper.find('form');
    form.simulate('submit', {
      preventDefault: () => {
        wrapper.setState({ gifCount: 20, gifs: new Array(20) });
      }
    } );
    expect(wrapper.state().gifCount).toBe(20)
    expect(wrapper.state().gifs.length).toBe(20)
  });

  it(`Should Match the snapshot`, () => {
    const tree = shallow(<SearchFor />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});