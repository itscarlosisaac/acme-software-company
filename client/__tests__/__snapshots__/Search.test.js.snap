// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFor/> - Search Component Should Match the snapshot 1`] = `
<Fragment>
  <form
    className="app__search"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search for a gif"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
